types/heft-jest/index.d.ts

```diff
@@ -32,5 +64,5 @@
  *
  * https://kulshekhar.github.io/ts-jest/user/test-helpers
  */
-declare function mocked<T>(item: T, deep?: false): mocked.MaybeMocked<T>;
-declare function mocked<T>(item: T, deep: true): mocked.MaybeMockedDeep<T>;
+function mocked<T>(item: T, deep?: false): mocked.MaybeMocked<T>;
+function mocked<T>(item: T, deep: true): mocked.MaybeMockedDeep<T>;
```

Comment:
Is this really necessary? Jest has `jest.mocked()` helper, is that not sufficient?

By the way, typings of `jest.mocked()` were reworked recently and the copy above is already outdated. For reference see: https://github.com/facebook/jest/blob/835a93666a69202de2a0429cd5445cb5f56d2cea/packages/jest-mock/src/index.ts#L34-L92, and: https://github.com/facebook/jest/blob/835a93666a69202de2a0429cd5445cb5f56d2cea/packages/jest-mock/src/index.ts#L1219-L1230

Reply:
So should this be changed to the following then?

```ts
declare global mocked: typeof import('@jest/globals').mocked
```

I’m not familiar with `heft-jest`. It’s just affected by the updates.

Reply:
I was looking through Heft’s docs. The way they use Jest does not work with `ts-jest`, so to have `mocked()` working they copied it to `@types/heft-jest`. Seemed like they are fine to deprecate `@types/heft-jest` after Heft will ship Jest v28, which has `jest.mocked()` builtin. I think that would be the best solution. See https://github.com/microsoft/rushstack/issues/3609